core/src/toga/style/pack.py

```diff
@@ -526,6 +532,56 @@ def _layout_column_children(self, node, available_width, available_height, scale
 
     def __css__(self):
         css = []
+        # display
+        if self.display == NONE:
+            css.append("display: none;")
+        else:
+            # if self.display != NONE, it must be pack; it will inherit
+            # the pack definition from the Toga stylesheet.
+            pass
+
+        # visibility
+        if self.visibility != VISIBLE:
+            css.append(f"visibility: {self.visibility}")
+
+        # direction
+        css.append(f"flex-direction: {self.direction.lower()};")
+
+        # alignment
+        if self.direction == ROW:
+            if self.alignment:
+                if self.alignment == LEFT:
+                    css.append("align-items: start;")
+                elif self.alignment == RIGHT:
+                    css.append("align-items: end;")
+                elif self.alignment == CENTER:
+                    css.append("align-items: center;")
+        else:
+            if self.alignment:
+                if self.alignment == TOP:
+                    css.append("align-items: start;")
+                elif self.alignment == BOTTOM:
+                    css.append("align-items: end;")
+                elif self.alignment == CENTER:
+                    css.append("align-items: center;")
+
+        # width
+        if self.width:
+            css.append(f"width: {self.width}px;")
+        elif self.direction == ROW and not self.flex:
+            css.append("flex: 0 0 0%;")
+
+        # height
+        if self.height:
+            css.append(f"width: {self.width}px;")
+        elif self.direction == COLUMN and not self.flex:
+            css.append("flex: 0 0 0%;")
+
+        # flex
+        if self.flex:
+            css.append(f"flex: {self.flex} 0 0%;")
```

Comment:
I believe I was relying on the fact that a flex definition will be ignored if width/height are defined. However, to avoid confusion, it's probably better to merge this into the width/height constructions above.

Reply:
Based on my tests, both `flex: 1 0 0` and `flex: 0 0 0` do have an effect in CSS even if there's an explicit width/height, so we'd better only generate one of them.